h2quic/client.go

```diff
@@ -178,7 +178,7 @@ func (c *client) RoundTrip(req *http.Request) (*http.Response, error) {
 		requestedGzip = true
 	}
 	// TODO: add support for trailers
-	endStream := !hasBody
+	endStream := !hasBody && req.Method != http.MethodConnect
```

Comment:
I removed this change and changed PR's description.

For the left `path` checking, does it make sense?